design-proposals/vpro-eim.md

```diff
@@ -0,0 +1,166 @@
+
+AMT does not provide any info about in progress operations, nor if a reboot is in-progress. However, The DM Resource Manager can fake the `IN_PROGRESS` statuses until the state in the DMT stack does not reflect the required operation.
+For example during the provisioning, the RM can keep the `IN_PROGRESS` state until the device is connected, move to an error status if the devices does not show up after a given threshold or can fake the `IN_PROGRESS` statuses using `Last*` timestamps exposed by MPS.
+
+The behavior of the resource manager is driven by the `DesiredPowerState` and `PowerCommandPolicy`, and internally keeps a
+timer between soft and hard when `ORDERED_OFF` is set in `PowerCommandPolicy`. The PowerStatus would be updated with this internal behavior in the DM RM, for example: "Soft power off succeeded", "Soft power off failed after 30s, forcing Hard power off", etc.
+
+Instead, UI or Users can fetch directly from MPS REST APIs (through the MT-GW), using device UUID, the following
+additional information such as:
+
+- `SoftwareVersion` - AMT version deployed in the device
+- `NetworkState` - the network state of the AMT interface
+- `IPAddress` - the IP address of the AMT interface
+- `PowerSource` - info about system power source (Battery/Power)
+- `Features` - AMT supported features and in-use
+- `LastUpdate` - Last change push to the device
+- `LastConnected` - Last time the device was connected
+- `LastSeen` - Last time the device was seen
+- `LastDisconnected` - Last time the device was disconnected
+
+Existing managers used for Onboarding & Provisioning need to be aware of the request to activate vPRO functionality.
+A new optional tink-action will be used to enable the feature see [vPRO/AMT/ISM devices activation](./vpro-device.md) and should not allow the removal or the deauth of a device if AMT was provisioned.
+
+Additionally, they will have to consider if the device supports the technology, if the User provided all the required configurations such as Domain/Provisioning certificate.
+
+For [automatic (aka nZTP) provisioning][nztp], the Provider resource `infra-onboarding` will not contain any field in
+the provider config regarding vPRO/AMT.
+
+The NA readiness should not consider the vPRO/AMT status, as it is not required to have the node fully operational.
+
+### MVP Requirements
+
+At the time of writing it is expected to support the following User flows:
+
+- User is able to configure the activation of Edge Node for vPro.
+- User is able to view the Hosts that have activated vPro.
+- User is able to select a Host and Power down a device.
+- User is able to select a Host and Power up a device.
+- The UI/API shall show that the device is Powered up/down.
+- User is able to power-cycle the Device.
+- User is able to verify if vPRO/ISM is supported or no;
+- User is able to recover the device if something goes wrong during the provisioning of the final OS;
```

Comment:
why do we want to take this to the MVP requirement. Recover term is overloaded. It can be as simple as starting from the beginning. I don't think our provisioning and onboarding workflows support - clean-up to a known working state to support recovery.